### DISCUSSION TITLE
[Feat]: Trigger event (autoplay:select) on autoplay index update

### DISCUSSION THREAD

**Discussion by Mitch-At-Work** [davidjerleke/embla-carousel#1048] [Ideas] [answered]
### Which variants of Embla Carousel are relevant to this feature request?

- [X] embla-carousel **(Core)**
- [ ] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [X] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-fade
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

### Feature description

An event emission on autoplay trigger (i.e. "autoplay:select") to notify listeners when an index change is being driven by the autoplay plugin.

We can detect an index change from the select event, but deriving the source of this index change is harder to track when resulting from the autoplay plugin.

### Additional Context

- PR: #1054.

As a library wrapper, we intend to provide users with both the event and event type (autoplay/focus/drag/click etc.) for index change callbacks, this includes passing through the underlying mouse/pointer events when an action such as drag is taken.

For drag events, we've built a custom plugin to handle this scenario: (https://github.com/microsoft/fluentui/blob/master/packages/react-components/react-carousel/library/src/components/pointerEvents.ts).

This event would simplify an autoplay callback equivalent:
![image](https://github.com/user-attachments/assets/fb8f2c07-b45c-4fd8-9cec-e85e64ee80e1)

I am happy to submit a PR with the above if this feature is approved.

### Before submitting

- [X] I've made research efforts and searched the documentation
- [X] I've searched for existing issues
- [X] I agree to follow this project's Contributing Guidelines for feature requests

**Comment by davidjerleke** [comment]
Hi @Mitch-At-Work,

Thank you for your suggestion. I won't have time to look into this at the time of writing but I don't see why not. I know there's a feature request about adding an option that controls the autoplay direction on the roadmap:

- #841

It will make it possible for autoplay to go backwards so with that in mind, you might want to consider alternative naming for `autoplay:next`.

**Comment by Mitch-At-Work** [comment]
PR raised here, please let me know if we need any additional testing or example updates to land:

- https://github.com/davidjerleke/embla-carousel/pull/1054

**Reply by davidjerleke** [reply]
@Mitch-At-Work I just released this feature in [8.4.0](https://github.com/davidjerleke/embla-carousel/releases/tag/v8.4.0). Thanks for your contribution.

**Reply by davidjerleke** [reply, marked as answer]
This feature is available from [v8.4.0](https://github.com/davidjerleke/embla-carousel/releases/tag/v8.4.0) and up.
